app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java

```diff
@@ -1,28 +1,70 @@
 package ar.com.wolox.android.example.ui.login;
 
+import java.util.ArrayList;
+import java.util.List;
 import javax.inject.Inject;
 import ar.com.wolox.android.example.utils.UserSession;
 import ar.com.wolox.wolmo.core.presenter.BasePresenter;
 
 /** <b>LoginPresenter</b>. */
 public class LoginPresenter extends BasePresenter<LoginView> {
 
-    private final UserSession userSession;
+    private UserSession userSession;
 
     @Inject
-    public LoginPresenter(final UserSession userSession) {
+    public LoginPresenter(UserSession userSession) {
         this.userSession = userSession;
     }
 
-    public void onLoginButtonClicked() {
-        // TODO
+    private final String emailRegex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
+
+    /**
+     * @param email email field, must not be empty
+     * @param password password field, must not be empty
+     */
+    public void onLoginButtonClicked(String email, String password) {
+        List<LoginError> errors = validateLoginCredentials(email, password);
+        if (!errors.isEmpty()) {
+            if (errors.contains(LoginError.EMPTY_EMAIL)) {
```

Comment:
We talked about this in private. And I find this possible solution to this issue, though if more errors will appear lot's of `if` are horrible.

````
    private enum ErrorCase {
        EMPTY_EMAIL {
            public void callAction(LoginView view) {
                view.invalidateEmptyEmail();
            }
        },
        EMPTY_PASSWORD {
            public void callAction(LoginView view) {
                view.invalidateEmptyPassword();
            }
        },
        INVALID_EMAIL {
            public void callAction(LoginView view) {
                view.invalidateEmailFormat();
            }
        };

        public abstract void callAction(LoginView view);
    }
```
You can replace every `if` with a loop and execute the callAction needed. (You can use other names if you want)